libraries/core_libs/consensus/src/transaction/transaction_queue.cpp

```diff
@@ -52,18 +72,51 @@ bool TransactionQueue::erase(const trx_hash_t &hash) {
   return true;
 }
 
-bool TransactionQueue::insert(std::shared_ptr<Transaction> &&transaction) {
-  assert(transaction);
-  if (hash_queue_.contains(transaction->getHash())) return false;
+bool TransactionQueue::insert(std::pair<std::shared_ptr<Transaction>, TransactionStatus> &&transaction,
```

Comment:
also I think there should be some check if the `last_block_number` for other status ten verified maybe just assert

Reply:
I dont understand this comment